### ISSUE TITLE
Make Spectral Normalization as HybridBlocks

### ISSUE THREAD

**Issue by stu1130** [apache/incubator-mxnet#13879]
## Description
The sn-gan example use customized Conv2D block. however, it would be better to make it as HybridBlock since all the calculation there have the Symbol API. One possible solution would be seperate the SNConv2D into Conv2D and Spectral Normailization HybridBlock similar to Batch [Normalization Block](https://mxnet.apache.org/api/python/gluon/nn.html#mxnet.gluon.nn.BatchNorm)
related to #13847

**Comment by stu1130** [comment]
@mxnet-label-bot add [Gluon, Good First Issue, Call for Contribution]
